tests/unit/s2n_resume_test.c

```diff
@@ -861,11 +886,65 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_config_free(config));
         }
 
+        /* Deserialized TLS1.2 session resumption for stateful session resumption using session id */
+        {
+            struct s2n_blob session_id_blob = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&session_id_blob, tls12_session_id, sizeof(tls12_session_id)));
+            struct s2n_stuffer session_id_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_init(&session_id_stuffer, &session_id_blob));
+            EXPECT_SUCCESS(s2n_stuffer_skip_write(&session_id_stuffer, sizeof(tls12_session_id) - S2N_TLS_SECRET_LEN));
+            /* The secret needs to be written to the ticket separately as it has a fixed length */
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&session_id_stuffer, test_master_secret.data, S2N_TLS_SECRET_LEN));
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(conn);
+
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_config_set_wall_clock(config, mock_time, NULL));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+            EXPECT_OK(s2n_deserialize_resumption_state(conn, NULL, &session_id_stuffer));
+
+            EXPECT_EQUAL(conn->actual_protocol_version, S2N_TLS12);
+            EXPECT_EQUAL(conn->secure.cipher_suite, &s2n_rsa_with_aes_128_gcm_sha256);
+
+            EXPECT_BYTEARRAY_EQUAL(test_master_secret.data, conn->secure.master_secret, S2N_TLS_SECRET_LEN);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_config_free(config));
+        }
+
+        /* Deserialized TLS1.3 session resumption for stateful session resumption not yet implemented */
+        {
+            struct s2n_blob session_id_blob = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&session_id_blob, tls13_session_id, sizeof(tls13_session_id)));
+            struct s2n_stuffer session_id_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_init(&session_id_stuffer, &session_id_blob));
+            EXPECT_SUCCESS(s2n_stuffer_skip_write(&session_id_stuffer, sizeof(tls13_session_id)));
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(conn);
+
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_config_set_wall_clock(config, mock_time, NULL));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+            EXPECT_ERROR_WITH_ERRNO(s2n_deserialize_resumption_state(conn, NULL, &session_id_stuffer),
```

Comment:
I'm sort of confused about why this test is necessary. It seems like it's just hitting the null_check() in s2n_tls13_deserialize_session_state. If we're testing to see if that function can take a null input, why not test that function explicitly rather than s2n_deserialize_resumption_state?